File: app/src/main/java/crane/com/mockappdemo/Application.java
package crane.com.mockappdemo;

import android.os.Environment;

import com.crane.mockapp.core.model.layouts.ProjectServiceFactory;

import java.io.File;

/**
 * Created by azhuravlev on 4/17/2017.
 */

public class Application extends android.app.Application {

    @Override
    public void onCreate() {
        super.onCreate();
        String path = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS), "MockAppDemo").getPath();
        ProjectServiceFactory.init(getApplicationContext(), path);
    }
}

File: app/src/main/java/crane/com/mockappdemo/Application.kt
package crane.com.mockappdemo

import android.app.Application
import android.os.Environment
import com.crane.mockapp.core.model.layouts.ProjectServiceFactory
import java.io.File

class Application : Application() {

    override fun onCreate() {
        super.onCreate()

        // Optionally you can override location of the local project's storage.
        // Default is /sdcard/Documents/MockApp
        val path =
            File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS), "MockAppDemo").path
        ProjectServiceFactory.init(this, path)
    }
}